### PR TITLE
handling mix enum

### PR DESCRIPTION
**Pull request by lilling**
hey @slavik57!

Good night!

### CODE REVIEW

File: src/enumValues.ts [hunk 1]
```diff
@@ -4,35 +4,22 @@ export class EnumValues {
...
-    }
+    const skinnyE = this.removeNumberKeys(e);
 
-    return Object.keys(e);
+    return Object.keys(skinnyE);
   }
 
-  static getValues(e: any) {
-    return this.getNames(e).map(name => e[name]) as number[];
+  private static removeNumberKeys(e: any) {
```
Comment (slavik57) [inline src/enumValues.ts]: Good eye for the pattern! But you can do it much simpler. You can delete this method and simply use:
```return Object.keys(e).filter(key => isNaN(key))``` in the implementation of the getNames

File: src/enumValues.ts [hunk 2]
```diff
@@ -4,35 +4,22 @@ export class EnumValues {
   }
 
   static getNames(e: any) {
-    if (this.isNumericEnum(e)){
-      return this.getObjectValues(e).filter(v => typeof v === "string") as string[];
-    }
+    const skinnyE = this.removeNumberKeys(e);
```
Comment (slavik57) [inline src/enumValues.ts]: Simply ```return Object.keys(e).filter(key => isNaN(key))```

**Review by slavik57 — CHANGES_REQUESTED** [review]
Awesome! You found a good pattern. Please view my comments =]
😢  It still does not support enums like these:
``` enum { A: 1, 2: B }```
But its better =]
Thanks

**Review by slavik57 — APPROVED** [review]

**Comment by lilling** [comment]
Hey fixed it.

I wrote `+key` because `isNaN` expecting a number as parameter.

and on your comment about enums with number as keys its not optional. see screen shot:
image
 

and you can edit your readme :-)

**Pull request merged**